### ISSUE TITLE
"goto" outside block corrupts execution stack

### ISSUE THREAD

**Issue by Quasilyte** [Quasilyte/goism#79]
Add tests in `tst/conformance` that fail due to this bug
and then fix it.

**Comment by Quasilyte** [comment]
```
--- FAIL: Test5Goto (0.02s)
	conformance_test.go:68: (testGotoScopes2 10)=>20 (want 10)
	conformance_test.go:68: (testGotoScopes3 10)=>-1 (want 10)
```

**Comment by Quasilyte** [comment]
```
--- FAIL: Test9For (0.04s)
	conformance_test.go:68: (testForScopes1 10)=>1 (want 10)
	conformance_test.go:68: (testForScopes2 10)=>1 (want 10)
```

**Comment by Quasilyte** [comment]
Test `(testGotoBackAndScopes 10)` kills Emacs.

**Comment by Quasilyte** [comment]
Fixed as a part of 1aa43965c71abce28cec467c9b3cab0838d38332.
